HACKING.md

```diff
@@ -13,14 +13,52 @@ DASHBOARD_API_V2=https://dashboard.staging.snapcraft.io/api/v2/
 LOGIN_URL=https://login.staging.ubuntu.com
 ```
 
-## Use custom GitHub Webhook URL
+## Snap automated builds
 
-When you run the project locally you probably want to link GitHub repos with your local environment. We recomend using something like [ngrok](https://ngrok.com/) to proxy your machine and update the following variable in your .env.local file:
+To use this functionality, you need to set up test accounts with **GitHub** and **Launchpad** to test.
+
+### GitHub
+
+Go and [register a new OAuth application](https://github.com/settings/applications/new) in your GitHub account. You can fill in the form however you want, and the only important detail is that you set the "Authorization callback URL" to `http://localhost:8004/github/auth/verify`.
+
+Once you've created the application, you should be given the client id and the client secret. Update your `.env.local` file with the your secrets:
+
+```bash
+GITHUB_CLIENT_ID=<Your client id>
+GITHUB_CLIENT_SECRET=<Your client secret>
+```
+
+#### Use custom GitHub Webhook URL
+
+When you run the project locally, you probably want to link GitHub repos with your local environment; however, GitHub wouldn't be able to reach your localhost. You can use something like [ngrok](https://ngrok.com/) to proxy your machine and update the following variable in your `.env.local` file:
```

Comment:
```suggestion
When you run the project locally, you probably want to link GitHub repos with your local environment; otherwise GitHub won't be able to reach your localhost. You can use a tool like [ngrok](https://ngrok.com/) to proxy to your machine and update the following variable in your `.env.local` file:
```